src/tests/Microsoft.PowerFx.Interpreter.Tests/IRTests.cs

```diff
@@ -64,6 +64,20 @@ public void RecordToRecordAggregateCoercionTest(string expr)
             Assert.IsNotType<ErrorValue>(result);
         }
 
+        [Theory]
+        
+        [InlineData("[{a:0},{a:\"3\",b:2}]", "Table({a:0,b:If(false,0)},{a:3,b:2})")]
+        [InlineData("[{a:1}, {a:true, b:2}]", "Table({a:1,b:If(false,0)},{a:1,b:2})")]
+        public void RecordToRecordAggregateCoercionDontDropFieldsTest(string expr, string expected)
```

Comment:
another test - is this recursive? (not sure the answer, but we should test either way)

[{b: {x:1}},
 {b: {x:"2", y:2}}]

Reply:
This fails. Which brings the question: in case this is allowed, what would be the deepest recursion level?